File: app/src/main/java/com/envirosoft/envirosense/services/DatabaseStorageService.java
package com.envirosoft.envirosense.services;

import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class DatabaseStorageService {
    public DatabaseStorageService(Context context) {
        if (isConnected(context)) {

        }
    }

    private boolean isConnected(Context context) {
        ConnectivityManager connMgr = (ConnectivityManager) context.getSystemService(Activity.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.isConnected();
    }
}

File: app/src/main/java/com/envirosoft/envirosense/service/DatabaseStorageService.kt
package com.envirosoft.envirosense.service

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager

object DatabaseStorageService {
  fun isConnected(context: Context): Boolean {
    val connMgr = context.getSystemService(Activity.CONNECTIVITY_SERVICE) as ConnectivityManager
    val networkInfo = connMgr.activeNetworkInfo
    return networkInfo != null && networkInfo.isConnected
  }
}
